### PR TITLE
[phrase matching] expose phrase condition

### PR DESCRIPTION
**Pull request by coszio**
- Adds a `Match::Phrase` variant to specify that a text must match by phrase.
- Adds a `phrase_matching` flag to the text index params so that text index registers positions.

Exposes it on rest and grpc

### CODE REVIEW

File: lib/segment/src/index/field_index/full_text_index/text_index.rs [hunk 1]
```diff
@@ -312,13 +312,26 @@ impl FullTextIndex {
         }
     }
 
+    /// Tries to parse a phrase query. If there are any unseen tokens, returns `None`
```
Comment (timvisee) [inline lib/segment/src/index/field_index/full_text_index/text_index.rs:281]: > If there are any unseen tokens, returns `None`

Is this correct?

File: tests/openapi/test_phrase_matching.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+
+@pytest.fixture(scope="module")
+def phrase_collection():
+    collection_name = setup_phrase_collection()
+    yield collection_name
+    drop_collection(collection_name)
+
+
+@pytest.mark.parametrize(
```
Comment (timvisee) [inline tests/openapi/test_phrase_matching.py:95]: Can we also add a case here that tests a duplicate word, like `"the the"`

**Review by timvisee — APPROVED** [review]
Please see all review remarks

File: lib/segment/src/index/field_index/full_text_index/text_index.rs (hunk 1 above)
Comment (coszio) [reply lib/segment/src/index/field_index/full_text_index/text_index.rs:281]: Yes, if a token hasn't been indexed, it means the query will return 0 results in this segment

File: tests/openapi/test_phrase_matching.py (hunk 2 above)
Comment (coszio) [reply tests/openapi/test_phrase_matching.py:95]: Added in https://github.com/qdrant/qdrant/pull/6670/commits/2bfb949be1ac47db533f3986b7d78e9c2d69e20b

Comment (timvisee) [reply tests/openapi/test_phrase_matching.py:95]: very very very very :clown_face:
